### PR TITLE
Change regex to match relation-types w/o quotes in Link header

### LINKED ISSUE #11629: -FollowRelLink does not work if Link header uses rel=next (no quotes) instead of rel="next"
# Steps to reproduce

Make a request to a server that returns `Link: <some url>; rel=next`.
This is valid according to the grammar: https://tools.ietf.org/html/rfc5988#section-5

> relation-types = relation-type
>                | <"> relation-type *( 1*SP relation-type ) <">
> relation-type  = reg-rel-type | ext-rel-type
> reg-rel-type   = LOALPHA *( LOALPHA | DIGIT | "." | "-" )
> ext-rel-type   = URI

The quoting is not required.

# Expected behavior

Should follow

# Actual behavior

Doesn't follow

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.0.0-rc.2
PSEdition                      Core
GitCommitId                    7.0.0-rc.2
OS                             Darwin 19.2.0 Darwin Kernel Version 19.2.0: Sat Nov  9 03:47:04 PST 2019; root:xnu-6153.61.1~20/RELEASE_X86_64
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by Marusyk**
# PR Summary

Change regex to match relation-types without quotes in Link header.

## PR Context

Fixes #11629 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

Please review,
Thank you in advance

### CODE REVIEW

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 1]
```diff
@@ -1829,7 +1829,7 @@ internal void ParseLinkHeader(HttpResponseMessage response, System.Uri requestUr
 
             // we only support the URL in angle brackets and `rel`, other attributes are ignored
             // user can still parse it themselves via the Headers property
-            string pattern = "<(?<url>.*?)>;\\s*rel=\"(?<rel>.*?)\"";
+            string pattern = "<(?<url>.*?)>;\\s*rel=(\"?)(?<rel>[^\"]*)\\2";
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs]: Looking definition in https://tools.ietf.org/html/rfc5988#section-5 I think it should be:
```
<(?<url>.*?)>;\s*rel=("?)(?<rel>.*?)\1[^\w-.]?
```

**Review by TravisEz13 — DISMISSED** [review]
Please fix tests failures and address @iSazonov 's comments

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs (hunk 1 above)
Comment (Marusyk) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs]: thanks but it won't match this case `Link: <https://example.com>; rel=preconnect`

Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs]: It seems your pattern is not for .Net. You can test in PowerShell with ([regex]"pattern").IsMatch("teststring").

Also please consider that if no quotes is used we should search a terminator that is not "alfanumeric, dash and point".

**Review by iSazonov — APPROVED** [review]

**Comment by TravisEz13** [comment]
It would be good to add a tests to make sure this doesn't regress

**Comment by iSazonov** [comment]
@TravisEz13 I do not find how we could add the test. We use AspNetCore for test and AspNetCore sends the header as-is.

**Comment by TravisEz13** [comment]
@iSazonov Thanks for the explanation.

**Comment by iSazonov** [comment]
@Marusyk Thanks for your contribution.

**Comment by Marusyk** [comment]
Thanks for review and help

**Pull request merged**